circuit/src/commonMain/kotlin/com/slack/circuit/CircuitContent.kt

```diff
@@ -45,15 +46,25 @@ internal fun CircuitContent(
   circuitConfig: CircuitConfig,
   unavailableContent: (@Composable (screen: Screen) -> Unit)?,
 ) {
-  val screenUi = circuitConfig.ui(screen)
+  val context = remember(screen, navigator, circuitConfig) { CircuitContext(circuitConfig) }
+  val eventListener =
+    remember(screen, context) {
+      circuitConfig.eventListenerFactory?.create(screen, context) ?: EventListener.NONE
+    }
+  DisposableEffect(screen) { onDispose { eventListener.dispose() } }
 
+  eventListener.onBeforeCreatePresenter(screen, navigator, context)
   @Suppress("UNCHECKED_CAST")
   val presenter = circuitConfig.presenter(screen, navigator) as Presenter<CircuitUiState>?
+  eventListener.onAfterCreatePresenter(screen, navigator, presenter, context)
+
+  eventListener.onBeforeCreateUi(screen, context)
+  val screenUi = circuitConfig.ui(screen)
+  eventListener.onAfterCreateUi(screen, screenUi, context)
 
   if (screenUi != null && presenter != null) {
-    val eventListener = circuitConfig.eventListenerFactory?.create(screen) ?: EventListener.NONE
     @Suppress("UNCHECKED_CAST")
-    CircuitContent(eventListener, presenter, screenUi.ui as Ui<CircuitUiState>)
+    CircuitContent(screen, eventListener, presenter, screenUi.ui as Ui<CircuitUiState>)
   } else if (unavailableContent != null) {
     unavailableContent(screen)
```

Comment:
I believe we should be calling `eventListener.onUnavailableContent` here

Reply:
good catch c47113c